### PR TITLE
[media-library] Prevent from crashing on Android 10

### LINKED ISSUE #5637: MediaLibrary.getAlbumAsync crashing on Android 10 (API Level 29)
### Environment

Android 10 (Pixel XL emulator with API level 29).
Also seen on production devices via the Play Store.

### Steps to Reproduce

Call `MediaLibrary.getAlbumAsync()`

### Expected Behavior

The album is returned

### Actual Behavior

The app crashes. Example error output:

```
09-13 13:31:07.621  6771  6816 E AndroidRuntime: Caused by: android.database.sqlite.SQLiteException: near "GROUP": syntax error (code 1 SQLITE_ERROR): , while compiling: SELECT bucket_id, bucket_display_name, COUNT(*) FROM files WHERE ((is_pending=0) AND (is_trashed=0) AND (volume_name IN ( 'external_primary' , '12f9-100d' ))) AND ((media_type != 0 AND bucket_display_name=?) /*)) ORDER BY */ GROUP BY bucket_id ORDER BY bucket_display_name
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:184)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.content.ContentProviderProxy.query(ContentProviderNative.java:423)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.content.ContentResolver.query(ContentResolver.java:944)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.content.ContentResolver.query(ContentResolver.java:880)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.content.ContentResolver.query(ContentResolver.java:836)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at expo.modules.medialibrary.MediaLibraryUtils.queryAlbum(MediaLibraryUtils.java:279)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at expo.modules.medialibrary.GetAlbum.doInBackground(GetAlbum.java:28)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at expo.modules.medialibrary.GetAlbum.doInBackground(GetAlbum.java:11)
09-13 13:31:07.621  6771  6816 E AndroidRuntime:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
```

### Reproducible Demo

Snack: https://snack.expo.io/@brsim/smiling-almond

Running this snack in the Expo app with the following emulator causes the app to crash:

```
  Name: Pixel_XL_API_29
  Device: pixel_xl (Google)
    Path: /Users/blake/.android/avd/Pixel_XL_API_29.avd
  Target: Google APIs (Google Inc.)
    Based on: Android API 29 Tag/ABI: google_apis/x86
  Skin: pixel_xl
  Sdcard: 100M
```

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolve #5637.

# How

We used the hack to add `GROUB BY` functionality to Android's cursors. Sadly, on Android 10, our hack is no longer valid (something has changed in the query builder). 
So I've created a new hack which achieves the same functionality as the old ones. 

# Test Plan

- test-suite passed (on Android 9 and 10)

### CODE REVIEW

**Review by mczernek — APPROVED** [review]

**Review by sjchmiela — APPROVED** [review]
Your solution doesn't look like a hack, are you sure it's a hack?

**Comment by mczernek** [comment]
> Your solution doesn't look like a hack, are you sure it's a hack?

It kinda is. It relies on some SQL database being used under the hood. It is, and was, always the case for any version of Android, but ContentProviders and ContentResolvers are intended to abstract data sharing functionality from any specific storage implementation.

On the other hand it isn't hack, because SQLite was is and probably will be there for eternity. So many people in so many places relies on it, that we can feel pretty secure. At least, what @lukmccall did is much much safer and cleaner than previous implementation 👏 👏

**Review by tsapeta — APPROVED** [review]

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java [hunk 1]
```diff
@@ -289,16 +289,18 @@ static void getExifFullInfo(Cursor cursor, Bundle response) throws IOException {
     response.putParcelable("exif", exifMap);
   }
 
-  static void queryAlbum(Context context, final String selection, final String[] selectionArgs, Promise promise) {
+  static void
+  queryAlbum(Context context, final String selection, final String[] selectionArgs, Promise promise) {
```
Comment (sjchmiela) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java]: <kbd>backspace</kbd> before `queryAlbum`

**Comment by sjchmiela** [comment]
Thank you, @mczernek, for the insightful comment

**Comment by impactcolor** [comment]
I setup my code like this:
  ```
      const albumResult = MediaLibrary.getAlbumAsync(currentOrderId);
        albumResult.then(result => {
          if(result) {
            resolve(result);
          } else {
            resolve(this.createPhotoAlbum(currentOrderId));
          }
        });
```
I'm still getting the crash after using hte above recommended fix.  Any ideas why?

**Comment by lukmccall** [comment]
Hi @impactcolor, could you open a new issue? It'll be easier to track it. 
Also, it would be nice if you provide me with more information. For example, logs from your device will be helpful.

**Pull request merged**